#!/usr/bin/env python

PACKAGE='auv_physics'
import roslib; roslib.load_manifest(PACKAGE)

from driver_base.msg import SensorLevels
from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

gen.add( "enable", bool_t, SensorLevels.RECONFIGURE_RUNNING, "Enable the thruster (Not implemented yet).", True )
gen.add( "invert", bool_t, SensorLevels.RECONFIGURE_RUNNING, "Invert the thruster direction after axis-thruster solution", False )
gen.add( "trim", double_t, SensorLevels.RECONFIGURE_RUNNING, "There's gotta be some trim here for me tonight", 0,     -100,  100 )
gen.add( "clamp_upper", double_t, SensorLevels.RECONFIGURE_RUNNING, "Max thruster output", 100,     0,  1000 )
gen.add( "clamp_lower", double_t, SensorLevels.RECONFIGURE_RUNNING, "Min thruster output", -100,    -1000,  0 )
gen.add( "use_clamp", bool_t, SensorLevels.RECONFIGURE_RUNNING, "If you want to clamp", False )
gen.add( "floor_mag", double_t, SensorLevels.RECONFIGURE_RUNNING, "Min thruster magnitude", 0, 0, 1000 )

################################################################################################################################
# Parameter Examples. Add your own parameters below
################################################################################################################################

# Valid types: bool_t, int_t, double_t, str_t

#          Name        Type   Reconfiguration level             Description                         Default Min Max
# gen.add( "my_param", int_t, SensorLevels.RECONFIGURE_RUNNING, "My very own dynamical parameter.", 10,     1,  100 )

# Example enum:
# size_enum = gen.enum([ gen.const("Small", int_t, 0, "A small constant"),
#                   gen.const("Medium", int_t, 1, "A medium constant"),
#                   gen.const("Large", int_t, 2, "A large constant"),
#                   gen.const("ExtraLarge", int_t, 3, "An extra large constant") ],
#                   "An enum to set size")

# gen.add("size", int_t, 0, "A size parameter which is edited via an enum", 1, 0, 3, edit_method=size_enum)

################################################################################################################################
################################################################################################################################


exit(gen.generate(PACKAGE, "thruster_model", "ThrusterModel"))
